### PR TITLE
Add workflow-multibranch-plugin

### PR DESCRIPTION
**Pull request by chriskilding**

### CODE REVIEW

**Review by jglick — DISMISSED** [review]
Something is wrong—not running PCT at all for the new plugin.

**Comment by jglick** [comment]
Working on it…

**Comment by jglick** [comment]
Fixed that, but now PCT is broken on the new plugin (which CI ought to now catch!):

```
Failed to execute goal on project workflow-multibranch: Could not resolve dependencies for project org.jenkins-ci.plugins.workflow:workflow-multibranch:hpi:2.21: Failure to find org.jenkins-ci.plugins.workflow:workflow-scm-step:jar:tests:2.11 in https://repo.jenkins-ci.org/public/ …
```

Seems that this is because https://github.com/jenkinsci/workflow-multibranch-plugin/pull/86 from @basil was merged months ago, yet never released. @dwnusbaum / @bitwiseman / et al. can this be released please?

File: bom-latest/pom.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
         <workflow-api-plugin.version>2.40</workflow-api-plugin.version>
         <workflow-cps-plugin.version>2.82</workflow-cps-plugin.version>
         <workflow-job-plugin.version>2.39</workflow-job-plugin.version>
+        <workflow-multibranch-plugin.version>2.21</workflow-multibranch-plugin.version>
```
Comment (dwnusbaum) [inline bom-latest/pom.xml:19]: I just released 2.22 which should fix the PCT issues:

```suggestion
        <workflow-multibranch-plugin.version>2.22</workflow-multibranch-plugin.version>
```

**Comment by jetersen** [comment]
```
- Update required: Pipeline: Supporting APIs (3.4) to be updated to 3.5 or higher
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:934)
	at hudson.PluginManager$2$1$1.run(PluginManager.java:548)
	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1131)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

`pct-configuration-as-code-2.176.x` failed because failFast 😓

**Comment by jglick** [comment]
Not sure offhand, would need to run the `local-test.sh` to see why.

**Comment by chriskilding** [comment]
rebuilding

**Review by jetersen — APPROVED** [review]
LGTM

File: sample-plugin/pom.xml [hunk 2]
```diff
@@ -96,12 +91,12 @@
...
+            <classifier>tests</classifier>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
-            <artifactId>workflow-job</artifactId>
-            <classifier>tests</classifier>
+            <artifactId>workflow-multibranch</artifactId>
             <scope>test</scope>
         </dependency>
```
Comment (jglick) [inline sample-plugin/pom.xml:101]:
```suggestion
        </dependency>
        <dependency>
            <groupId>org.jenkins-ci.plugins.workflow</groupId>
            <artifactId>workflow-multibranch</artifactId>
            <classifier>tests</classifier>
            <scope>test</scope>
        </dependency>
```

**Review by jglick — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Would be a great addition to the BOM

**Comment by jetersen** [comment]
Merging once build passes :)

**Comment by chriskilding** [comment]
rebased, build passing :)

**Pull request merged**